components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt

```diff
@@ -49,11 +60,74 @@ private fun Suggestions(
     onSuggestionClicked: (AwesomeBar.Suggestion) -> Unit
 ) {
     suggestions.forEach { suggestion ->
+        Suggestion(suggestion, onSuggestionClicked)
+    }
+}
+
+@Composable
+private fun Suggestion(
+    suggestion: AwesomeBar.Suggestion,
+    onSuggestionClicked: (AwesomeBar.Suggestion) -> Unit
+) {
+    Row(
+        modifier = Modifier
+            .clickable { onSuggestionClicked(suggestion) }
+            .padding(start = 16.dp, top = 8.dp, bottom = 8.dp)
+    ) {
+        SuggestionIcon(
+            icon = suggestion.icon,
+            modifier = Modifier.align(Alignment.CenterVertically)
+        )
+        SuggestionTitleAndDescription(
+            title = suggestion.title,
+            description = suggestion.description,
+        )
+    }
+}
+
+@Composable
+private fun SuggestionTitleAndDescription(
+    title: String?,
+    description: String?,
+) {
+    Column {
         Text(
-            text = suggestion.title ?: "",
-            modifier = Modifier.padding(8.dp)
+            text = title ?: "",
+            fontSize = 15.sp,
+            maxLines = 1,
+            modifier = Modifier
                 .fillMaxWidth()
-                .clickable { onSuggestionClicked(suggestion) }
+                .padding(start = 8.dp, end = 8.dp)
         )
+        if (description?.isNotEmpty() == true) {
+            CompositionLocalProvider(LocalContentAlpha provides ContentAlpha.medium) {
+                Text(
+                    text = description,
+                    fontSize = 12.sp,
+                    maxLines = 1,
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(start = 8.dp, end = 8.dp)
+                )
+            }
+        }
     }
 }
+
+@Composable
+private fun SuggestionIcon(
+    icon: Bitmap?,
+    modifier: Modifier
+) {
+    if (icon == null) {
```

Comment:
Instead of having this check here making this imperative, you could add the null check for `SuggestionIcon` inside the row of `Suggestion` composable.

Reply:
Yeah, I moved the code to `SuggestionIcon` in order to simplify `Suggestion`, which was growing a lot. If only `Image` is left here then I can fold everything into `Suggestion` and it's a big method again. 🤔

Reply:
It still makes sense to have every small component separately even if it is just an image with a specific modifier. For this case I just meant you could have something like this: https://github.com/mozilla-mobile/android-components/blob/dc525f1096ccb8639cc06e70851242945cdbcc72/components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt#L78